opt_adapt/opt.py

```diff
@@ -171,8 +249,14 @@ def minimise(
         dJ_ = None if it == 1 else op.dJdm_progress[-1]
         if order == 1:
             args = (u_plus, u_, dJ_)
+        elif order == 2:
+            if method == "BFGS":
+                B = None
+                args = (u_plus, u_, dJ_, B)
+            if method == "newton":
+                args = (u_plus)
```

Comment:
It would be better to have the same `args` for both methods, even if some of them are unused.